x/crosschain/keeper/evm_hooks.go

```diff
@@ -78,47 +80,71 @@ func (k Keeper) ProcessLogs(
 	if connectorZEVMAddr == (ethcommon.Address{}) {
 		return fmt.Errorf("connectorZEVM address is empty")
 	}
+	gatewayAddr := ethcommon.HexToAddress(system.Gateway)
 
+	// read the logs and process inbounds from emitted events
+	// run the processing for the v1 and the v2 protocol contracts
 	for _, log := range logs {
-		eventZrc20Withdrawal, errZrc20 := ParseZRC20WithdrawalEvent(*log)
-		eventZetaSent, errZetaSent := ParseZetaSentEvent(*log, connectorZEVMAddr)
-		if errZrc20 != nil && errZetaSent != nil {
-			// This log does not contain any of the two events
-			continue
+		if !crypto.IsEmptyAddress(gatewayAddr) {
```

Comment:
v1 and v2 events will be both parsed sequentially but for v2 it can only happen if the gateway is defined in the protocol otherwise events can't be parsed